bambi/models.py

```diff
@@ -746,9 +746,9 @@ def __init__(self, name, data, categorical=False, random=False, prior=None,
 
         # For the sake of computational efficiency (i.e., to avoid lots of
         # large matrix multiplications in the backends), invert the dummy-
-        # coding process and represent binary categoricals as a vector of
+        # coding process and represent full-rank dummies as a vector of
         # indices into the coefficients.
-        if data.shape[1] > 1 and ((data == 0) | (data == 1)).all():
+        if constant and data.shape[1] > 1 and ((data == 0) | (data == 1)).all():
```

Comment:
Your explanation re: the n-1 coding makes sense. Re: sparse matrices, we could certainly represent represent the data as sparse scipy matrices, but that wouldn't really save us any work in terms of the backends, because I don't want to deal with sparse matrices in the backends (and AFAIK, Stan doesn't even support them). So we would still need to collapse the matrices into the index-based representation.

The other issue is that we use patsy to generate the predictor matrix, and unless patsy can implicitly create and return sparse matrices, we're already committed to representing the dense matrix in memory at least transiently, so we wouldn't really gain anything unless you think it will help with the scaling calculations. But so far it doesn't seem like anything other than the backend computations is slowing things down.